File: serveur/app/models/nutrition_goal.py
# app/models/nutrition_goal.py
#
# Nutrition V1 (2026-06-12, cf. docs/NUTRITION_DESIGN.md §3.7).
# Cibles quotidiennes kcal + macros. Type A user-scoped.
# La cible active un jour J = celle avec le plus grand effective_from <= J
# (les stats passees comparent chaque jour a la cible active CE jour-la).
# day_kind UPPER_CASE (politique 11) : ALL en v1, extension future par type de
# jour (D3) sans migration. default=ALL legitime (politique 10 : cas majoritaire v1).
# Vue hebdo (D8) : pas d'entite dediee, le client compare somme(jours) a cible x 7.

from sqlalchemy import Column, Integer, String, Float, ForeignKey, DateTime
from app.database import Base
import uuid


class NutritionGoal(Base):
    __tablename__ = "nutrition_goals"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    uuid = Column(String, nullable=False, unique=True, default=lambda: str(uuid.uuid4()))

    effective_from = Column(String, nullable=False, index=True)  # "YYYY-MM-DD" (convention projet)
    day_kind = Column(String, nullable=False, default="ALL")     # ALL (v1) ; extension future D3

    kcal = Column(Float, nullable=False)
    protein_g = Column(Float, nullable=False)
    carbs_g = Column(Float, nullable=False)
    fat_g = Column(Float, nullable=False)

    updated_at = Column(DateTime(timezone=True), nullable=True)

    def __repr__(self):
        return (
            f"NutritionGoal(id={self.id}, user_id={self.user_id}, uuid='{self.uuid}', "
            f"effective_from='{self.effective_from}', kcal={self.kcal})"
        )

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/NutritionGoal.kt
package com.example.sportapp.core.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601

/**
 * Nutrition A1 (2026-06-17) — portage Android du modèle serveur `nutrition_goals`
 * (cf. serveur/app/models/nutrition_goal.py). Cibles quotidiennes kcal + macros.
 * Type A user-scoped.
 *
 * La cible active un jour J = celle avec le plus grand `effective_from` <= J.
 * `day_kind` UPPER_CASE (politique 11) : ALL en v1.
 */
@Entity(
    tableName = "nutrition_goals",
    indices = [
        Index(value = ["uuid"], unique = true),
        Index(value = ["user_id"]),
        Index(value = ["effective_from"]),
    ]
)
data class NutritionGoal(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,

    @ColumnInfo(name = "effective_from") val effectiveFrom: String,   // "YYYY-MM-DD"
    @ColumnInfo(name = "day_kind") val dayKind: String = "ALL",       // ALL (v1)

    val kcal: Float,
    @ColumnInfo(name = "protein_g") val proteinG: Float,
    @ColumnInfo(name = "carbs_g") val carbsG: Float,
    @ColumnInfo(name = "fat_g") val fatG: Float,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
